### PR TITLE
0.01 - Initial commit of project skeleton

### PR DESCRIPTION
**Pull request by Bernard-hu**

### CODE REVIEW

File: src/main/resources/schema.sql
Comment (dmytropanov-sa) [inline src/main/resources/schema.sql:1]: What do you think if move this schema db to migrations, so we can maintain db schema. For this you can add in build.gradle implementation 'org.flywaydb:flyway-core', and create src/main/resources/db/migration folder appropriately for migrations.

**Review by dmytropanov-sa — CHANGES_REQUESTED** [review]
Everything is Great! Nice job! Even JUnit, WebSecurity and Log4J is here....
Just several comments around adding migrations and managing DB secrets...

File: src/main/resources/schema.sql
Comment (Bernard-hu) [reply src/main/resources/schema.sql:1]: Good point, I added the dependency, and moved the schema file. Thank you for your help!

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM openjdk:17
```
Comment (dmytropanov-sa) [inline Dockerfile]: Lets update java to v 21!

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	id 'java'
+	id 'org.springframework.boot' version '3.2.2'
+	id 'io.spring.dependency-management' version '1.1.4'
+}
+
+group = 'com.epam'
+version = '0.0.1-SNAPSHOT'
+
+java {
+	sourceCompatibility = '17'
```
Comment (dmytropanov-sa) [inline build.gradle]: Lets update Java v to 21!

File: db/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+FROM postgres:16
+ENV POSTGRES_USER=postgres
+ENV POSTGRES_PASSWORD=1234
```
Comment (dmytropanov-sa) [inline db/Dockerfile]: What do you think to move pass to env variable and do not store in version control system?

File: db/migration/schema.sql
Comment (dmytropanov-sa) [inline db/migration/schema.sql:1]: Please find post about flywaydb migrations configurations here: https://medium.com/swlh/introduction-of-flyway-with-spring-boot-d7c11145d012, and adjust code please.

**Review by dmytropanov-sa — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Bernard-hu) [reply Dockerfile]: I altered the version.

File: build.gradle (hunk 2 above)
Comment (Bernard-hu) [reply build.gradle]: I altered the version.

File: db/migration/schema.sql
Comment (Bernard-hu) [reply db/migration/schema.sql:1]: Thank you for the information. Renamed schema.sql, and set extra property to the application.properties file

File: db/Dockerfile (hunk 3 above)
Comment (Bernard-hu) [reply db/Dockerfile]: Great advise, it have been fixed by now in the 0.04 commit. Passed the username as well.

**Review by dmytropanov-sa — APPROVED** [review]
Nice Job! Really Great.

**Pull request merged**